src/Common/tests/System/PlatformDetection.cs

```diff
@@ -18,9 +18,10 @@ public static partial class PlatformDetection
         // do it in a way that failures don't cascade.
         //
 
-        public static bool IsUap => IsWinRT || IsNetNative;
+        public static bool IsUap => IsInAppContainer || IsNetNative;
```

Comment:
>@davidsh any reason why you implemented IsUap like that? If we fix the behavior to what I was suggesting in the above link and what Tarek is suggesting here, would we break some new tests of yours?

Yes, I implemented that way in order to the "IsUap" to work.  See issue #19931.  "IsUap" needs to return true whether it is 'uap' or 'uap-aot' configuration and regardless of the current implementation behavior of running 'uap-aot' tests in a non app-container way.

If you change this, you might break our tests.  So, you will need validate this against all uap and uap-aot flavors.